components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/feature/OriginVerifierFeature.kt

```diff
@@ -18,14 +17,16 @@ import mozilla.components.feature.customtabs.store.VerificationStatus.FAILURE
 import mozilla.components.feature.customtabs.store.VerificationStatus.PENDING
 import mozilla.components.feature.customtabs.store.VerificationStatus.SUCCESS
 import mozilla.components.feature.customtabs.verify.OriginVerifier
+import mozilla.components.service.digitalassetlinks.RelationChecker
 
 class OriginVerifierFeature(
-    private val httpClient: Client,
     private val packageManager: PackageManager,
-    @VisibleForTesting internal val apiKey: String?,
+    private val relationChecker: RelationChecker,
     private val dispatch: (CustomTabsAction) -> Unit
 ) {
 
+    private var cachedVerifier: Triple<String, Int, OriginVerifier>? = null
```

Comment:
It's not obvious to be why we cache the last verifier. If you open a different PWA, the last verifier would be dropped, IIUC?

Reply:
This is a minor optimization to just handle opening the same PWA again. To keep the caching simple, we just drop it when a different PWA is opened.